File: src/jray/math/intersections/PointCube.java
package jray.math.intersections;

import jray.common.Vect3;

public class PointCube {
	/**
	 * Returns true if, and only if, the point <code>point</code> is enclosed in the given cube. 
	 * 
	 * @param cubeCenter
	 * @param cubeWidthHalf
	 * @param point
	 * @return
	 */
	public static boolean encloses(Vect3 cubeCenter, double cubeWidthHalf, Vect3 point){
		return Math.abs(cubeCenter.data[0]-point.data[0])<cubeWidthHalf&&
	       Math.abs(cubeCenter.data[1]-point.data[1])<cubeWidthHalf&&
	       Math.abs(cubeCenter.data[2]-point.data[2])<cubeWidthHalf;
	}
}

File: JRayXLib/JRayXLib/Math/intersections/PointCube.cs
using JRayXLib.Shapes;

namespace JRayXLib.Math.intersections
{
    public class PointCube
    {
        /**
     * Returns true if, and only if, the point <code>point</code> is enclosed in the given cube. 
     * 
     * @param cubeCenter
     * @param cubeWidthHalf
     * @param point
     * @return
     */

        public static bool Encloses(Vect3 cubeCenter, double cubeWidthHalf, Vect3 point)
        {
            return System.Math.Abs(cubeCenter.X - point.X) < cubeWidthHalf &&
                   System.Math.Abs(cubeCenter.Y - point.Y) < cubeWidthHalf &&
                   System.Math.Abs(cubeCenter.Z - point.Z) < cubeWidthHalf;
        }
    }
}